iceoryx_utils/include/iceoryx_utils/internal/concurrent/lockfree_queue/index_queue.hpp

```diff
@@ -85,6 +85,12 @@ class IndexQueue
     /// @return index if the queue was full, nullopt otherwise
     cxx::optional<ValueType> popIfFull() noexcept;
 
+    /// @brief pop an index from the queue in FIFO order if the queue contains
+    ///        at least  a specified number number of elements
+    /// @param size the number of elements needed to successfully perform the pop
+    /// @return index if the queue contains size elements, nullopt otherwise
+    cxx::optional<ValueType> popIfSizeIsAtLeast(uint64_t size) noexcept;

```

Comment:
as cvasfi already pointed out with some local variable, the parameters should also be const. I leave it to you to check where it applies, to not litter the review with comments regarding const

Reply:
Hm....I do not like making value parameters const, you know that. It is not useful for the caller and limits the implementation. Note that from my understanding the optimizer does not need it, the compiler can see anyway whether a value is actually written or not (it will not rely on const, since there are ways to circumvent it). Meaning a potential optimization advantage does not depend on const declarations.

But I will take a look.
Any oversight on references will be fixed of course.